test/integration/api/datasets.js

```diff
@@ -2030,6 +2030,209 @@ describe('datasets and entities', () => {
             }));
       }));
 
+      it('should ignore a saveto incorrrectly placed on a bind on a structural field', testService(async (service) => {
+        const alice = await service.login('alice');
+        const xml = `<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:entities="http://www.opendatakit.org/xforms">
+          <h:head>
+            <model entities:entities-version='2022.1.0'>
+              <instance>
+                <data id="validate_structure">
+                  <name/>
+                  <age/>
+                  <group>
+                    <inner_field/>
+                  </group>
+                  <meta>
+                    <entity dataset="things" id="" create="1">
+                      <label/>
+                    </entity>
+                  </meta>
+                </data>
+              </instance>
+              <bind nodeset="/data/name" type="string" entities:saveto="prop1"/>
+              <bind nodeset="/data/group" entities:saveto="prop2"/>
+              <bind nodeset="/data/group/inner_field" type="string" entities:saveto="prop3"/>
+            </model>
+          </h:head>
+        </h:html>`;
+        await alice.post('/v1/projects/1/forms')
+          .send(xml)
+          .set('Content-Type', 'application/xml')
+          .expect(200);
+        await alice.get('/v1/projects/1/forms/validate_structure/draft/dataset-diff')
+          .expect(200)
+          .then(({ body }) => {
+            const { properties } = body[0];
+            properties.length.should.equal(2);
+            properties[0].name.should.equal('prop1');
+            properties[1].name.should.equal('prop3');
+          });
+        await alice.post('/v1/projects/1/forms/validate_structure/draft/publish')
+          .expect(200);
+        await alice.get('/v1/projects/1/datasets/things')
+          .expect(200)
+          .then(({ body }) => {
+            body.name.should.be.eql('things');
+            const { properties } = body;
+            properties.length.should.equal(2);
+            properties[0].name.should.equal('prop1');
+            properties[1].name.should.equal('prop3');
+          });
+      }));
+
+      it.skip('should not break when a saveto/bind is on a field inside a repeat group', testService(async (service, container) => {
```

Comment:
forgot to remove `skip`?

this is about Submission parsing, right?

Reply:
this is me leaving a trail of my thoughts in a confusing manner
- towards the beginning of my work on this issue, i wrote this test checking that you could have binds on a field in a repeat, and the submission parsing would pick up the last value in the repeat.
- i decided i could get more aggressive about checking the form definition up front. with that change, this test no longer passes. i wasn't ready to delete it, though.
- i'll delete it unless for some reason we _do_ want to let people upload form definitions with binds on fields inside of repeats.